### PR TITLE
[interfaces] Updates to support MDS switches

### LINKED ISSUE #548: nxos_interfaces should support fc interfaces
Opening this issue to add support for fc interfaces for nxos_interfaces module

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

##### SUMMARY
Several assorted changes to support NX-OS on MDS switches.
Fixes #548 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
nxos_interfaces.py

### CODE REVIEW

**Comment by NilashishC** [comment]
cc @srbharadwaj

File: plugins/modules/nxos_interfaces.py [hunk 1]
```diff
@@ -78,18 +78,29 @@
         description:
         - Manage Layer2 or Layer3 state of the interface. Applicable for Ethernet
```
Comment (KB-perByte) [inline plugins/modules/nxos_interfaces.py:79]:
```suggestion
        - Manage Layer2 or Layer3 state of the interface (Applicable for Ethernet Interfaces) Manage port mode for fiber channel Interfaces. Applicable for Ethernet
```

File: plugins/modules/nxos_interfaces.py [hunk 2]
```diff
@@ -99,12 +110,19 @@
...
         type: bool
       fabric_forwarding_anycast_gateway:
         description:
         - Associate SVI with anycast gateway under VLAN configuration mode. Applicable
           for SVI interfaces only.
+        - This option is not applicable for NX-OS on MDS Switches.
         type: bool
+      analytics:
+        description:
+          - This option is applicable only for NX-OS on MDS Switches.
```
Comment (KB-perByte) [inline plugins/modules/nxos_interfaces.py:123]:
```suggestion
          - This option is applicable only for NX-OS on MDS Switches. Enable analytics for FC interfaces
```

File: plugins/module_utils/network/nxos/config/interfaces/interfaces.py [hunk 3]
```diff
@@ -70,12 +73,29 @@ def get_interfaces_facts(self, data=None):
```
Comment (rubenribeiro) [inline plugins/module_utils/network/nxos/config/interfaces/interfaces.py:73]: Returns 'nexus' facts by default. May want to pass 'mds' value when instantiating  Facts object.

**Pull request closed without merging**